src/Test/Inspection.hs

```diff
@@ -155,7 +157,12 @@ These convenience functions create common test obligations directly.
 infix 9 ===
 
 -- | Declare two functions to be equal, but ignoring
--- type lambdas, type arguments and type casts (see 'EqualTo')
+-- hpc ticks, type lambdas, type arguments and type casts (see 'EqualTo').
+--
+-- Note: If the code being inspected contains if- or case-expression,
+-- the generated Core can contain trivial branching, even with @-O2@ option.
+-- To test the optimised code correctly, we recommend turning off @-fhpc@ option,
+-- and testing against libraries compiled /without/ @-fhpc@ (or equivalent) option.
```

Comment:
Thanks! I’d like to make the “hpc tick” comment less prominent; I don’t want to scare people that don't know what these are away from the library :-)

```suggestion
-- type lambdas, type arguments, type casts and hpc ticks (see 'EqualTo').
-- Note that `-fhpc` can prevent some optimizations; build without for more reliable analysis.
```